tornjak-frontend/src/components/agent-list.tsx

```diff
@@ -58,34 +75,31 @@ class AgentList extends Component {
       this.TornjakApi.refreshLocalSelectorsState(this.props.agentworkloadSelectorInfoFunc);
       this.TornjakApi.populateLocalAgentsUpdate(this.props.agentsListUpdateFunc, this.props.tornjakMessageFunc);
       this.TornjakApi.populateLocalTornjakServerInfo(this.props.tornjakServerInfoUpdateFunc, this.props.tornjakMessageFunc);
-      if(this.props.globalTornjakServerInfo !== "") {
+      if (this.props.globalTornjakServerInfo !== "" && JSON.stringify(this.props.globalTornjakServerInfo) !== '{}') {
```

Comment:
Comparison with empty string not needed.
also this.props.globalTornjakServerInfo if known as object could be compared to the keys it has.

An alternative could be:
```
if (this.props.globalTornjakServerInfo && this.props.globalTornjakServerInfo && Object.keys(this.props.globalTornjakServerInfo).length) {
```